.github/workflows/post-pr-comment.yml

```diff
@@ -0,0 +1,30 @@
+name: Post pull-request comment
+on:
+  workflow_run:
+    workflows: ["Query help preview"]
+    types:
+      - completed
+
+permissions:
+  pull-requests: write
+
+jobs:
+  post_comment:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Download artifact
+        run: gh run download "${WORKFLOW_RUN_ID}" --repo "${GITHUB_REPOSITORY}" --name "comment"
+        env:
+          GITHUB_TOKEN: ${{ github.token }}
+          WORKFLOW_RUN_ID: ${{ github.event.workflow_run.id }}
+      - run: |
+          PR="$(grep -o '^[0-9]\+$' pr.txt)"
```

Comment:
Can we get the PR number in the workflow_run payload? 
From a test job it looks like `github.event.workflow_run.pull_requests[0].number` will have this information, which will simplify things here. I cannot tell the rules for when the `pull_requests` array is populated.

Reply:
It seems that way, and I tried to use it before. Unfortunately the field is not populated for pull request from forks. The checks API documentation is full of notes telling this, however, there is no explanation why.

Reply:
Forks! How sad. Ok then you've got the best way I can think of :)